sound/soc/sof/pcm.c

```diff
@@ -707,7 +707,14 @@ int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd, struct snd_pcm_hw_pa
 		}
 		break;
 	case SOF_DAI_INTEL_ALH:
-		/* do nothing for ALH dai_link */
+		/*
+		 * Sdw DSM may set dai channel count to 4 but
+		 * only create 2 channel catpure stream for user.
+		 * add fixup to set dai channel count to component
+		 * driver to set up hardware.
```

Comment:
Maybe the comment could be more generic, no need to mention the specific DSM here, something like the dai could run with different channels comparing with the user front end.

Reply:
expand SDW fully to say soundwire or ALH and generalize the commit title and message to say "Add be_hw_params_fixup() for ALH" or something like that

Reply:
Thanks very much .